### PR TITLE
Add templating to support dualstack ipv6 primary

### PR DESCRIPTION
**Pull request by tylerschultz**
### What this PR does / why we need it
This PR makes it possible to deploy dualstack, primary ipv6 management clusters. 

- Various Kuberntes components don't bind to the correct address in the
  case ipv6 primary dualstack i.e it doesn't bind to the ipv6 interface.
  This will make it explictly bind to the ipv6 interface.
- The apiServer doesn't set the advertise address correctly in the case
  of ipv6 primary dualstack. This will explicitly advertise the
  VSPHERE_CONTROL_PLANE_ENDPOINT.
- Set the node-ip to '::' to make kubelet prefer the ipv6 address on the node.

### Which issue(s) this PR fixes

Related to #616

### Describe testing done for PR

Created vSphere management cluster to verify change.
Unit test coverage for ytt templating overlays.

### Release note

```release-note
Experimental support for dualstack primary ipv6 (TKG_IP_FAMILY "ipv6,ipv4")
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by tylerschultz** [comment]
Also related is this feature gate PR: #1013

**Review by adobley — APPROVED** [review]

**Review by nicolehanjing — APPROVED** [review]

File: pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay.yaml [hunk 1]
```diff
@@ -209,6 +209,36 @@ spec:
       dns:
         imageRepository: #@ kubeadm_image_repo(bomDataForK8sVersion.kubeadmConfigSpec.dns.imageRepository)
         imageTag: #@ bomDataForK8sVersion.kubeadmConfigSpec.dns.imageTag
+      #@ if data.values.TKG_IP_FAMILY == "ipv6,ipv4":
```
Comment (vuil) [inline pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay.yaml:212]: I'd expect this to produce some clustergen diffs. Are we missing use cases generated to exercise this?

Comment (tylerschultz) [reply pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay.yaml:212]: Ah, makes sense. We will look at getting some reasonable clustergen diffs.

Comment (tylerschultz) [reply pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay.yaml:212]: clustergen has apparently been broken for a while now, there are problems with some of the existing constraints. We'll be submitting a separate PR to get them working correctly again.

Comment (tylerschultz) [reply pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay.yaml:212]: https://github.com/vmware-tanzu/tanzu-framework/pull/1073

**Comment by mcwumbly** [comment]
@tylerschultz, I'm adding `do-not-merge/hold` label on this PR for now since you mentioned you'd like to do further validation after the recent rebase to resolve conflicts. Let myself and @vuil know here when you think it's good to go again.

**Comment by tylerschultz** [comment]
We've reviewed the changes and are now comfortable with merging this PR. The changes that gave us pause were newly introduced settings related to AVI. These changes should not conflict with dualstack settings. We have an upcoming work related to AVI dualstack integration where we'll cover this in depth.

**Review by mcwumbly — APPROVED** [review]

**Review by vuil — APPROVED** [review]
lgtm thanks.

**Pull request merged**